demo/client/backend/src/fabricClient.js

```diff
@@ -0,0 +1,477 @@
+//const userName       = 'Admin';  //  hardcoded !
+const org1Affiliation = 'org1';  //  different from MSP;  used in CA
+
+//  affiliation is used in CA;  to add to the list of affiliation, customize
+//  fabric-ca-server-config.yaml and pass in as cafiles in start command.
+//  If not, use 'org1' and 'org2' - case sensitive, which is defined in
+// the default  fabric-ca-server-config.yaml.
+
+/////////////////////////////  end of input  ///////////////////////////
+
+//  global variables loaded initially
+var configdata;
+var mspDir;
+var connectionProfile;
+var wallet;
+var adminUserName;
+var secret;
+var bLocalHost;  //  used in setUserContext
+
+
+//  other global variables
+var strResponse;
+var gateway;
+var network;
+var contract;
+
+//  exported functions
+async function connectToNetwork(channelName, chaincodeID) {
+    console.log (">>> In function, connectToNetwork: channelName, chaincodeID = ", channelName, chaincodeID);
+
+    await beginNetworkConnect();
+
+    try {
+      gateway = await new Gateway();
+    }  catch (error) {
+      console.log(`Error creating new gateway`);
+      console.log(error.stack);
+    }
+
+    await gateway.connect(connectionProfile,
+        	  {
+              wallet,
+              identity:            adminUserName,
+              discovery:           { enabled: false, localhost: bLocalHost },
+              eventHandlerOptions: { strategy: null }
+      		  });
+    console.log ("Connected to gateway");
+    network =  await gateway.getNetwork (channelName);
+    console.log ("Connected to channel, " + channelName);
+
+    contract =  await network.getContract (chaincodeID);
+    console.log ("Connected to chaincode, ", chaincodeID);
+    console.log ('...........................');
+
+    await endNetworkConnect();
+    console.log ("<<< Exit: function, connectToNetwork");
+}
+
+async function getRegisteredUsers() {
+    console.log (">>>  Function getRegisteredUsers... ")
+
+    var client, fabric_ca_client, idService;
+
+    try {
+       client = gateway.getClient();
+    } catch (error) {
+       console.log(`Error in call to gateway.getClient: ${error}`);
+       console.log(error.stack);
+       throw (`Error in call to gateway.getClient: ${error}`);
+    }
+
+    try {
+      fabric_ca_client = client.getCertificateAuthority();
+    } catch (error) {
+      console.log(`Error in call to client.getCertificateAuthority: ${error}`);
+      console.log(error.stack);
+      throw (`Error in call to client.getCertificateAuthority: ${error}`);
+    }
+
+    try {
+       idService = fabric_ca_client.newIdentityService();
+    } catch (error) {
+       console.log(`Error in call to fabric_ca_client.newIdentityService: ${error}`);
+       console.log(error.stack);
+       throw (`Error in call to fabric_ca_client.newIdentityService: ${error}`);
+    }
+
+    var adminIdentity = await gateway.getCurrentIdentity();  //  which is org1admin
+
+    //  user must be a hf.registrar
+    try {
```

Comment:
is the try really needed round so much code or is it only needed for the `idService.getAll` call?